src/game/common/sparsematchfinder.h

```diff
@@ -17,18 +17,97 @@
 #include "always.h"
 #include <map>
 
-template<typename K, typename T> class SparseMatchFinder
+template<typename T, typename K> class SparseMatchFinder
 {
     class MapHelper
     {
     public:
         bool operator()(const K &left, const K &right) const { return left == right; }
     };
 
+    static int Count_Condition_Intersection(const K &key1, const K &key2) { return key1.Count_Intersection(key2); }
+    static int Count_Condition_Inverse_Intersection(const K &key1, const K &key2)
+    {
+        return key1.Count_Inverse_Intersection(key2);
+    }
+
 public:
-    // TODO implement more wrappers as they are determined required.
-    T &operator[](const K &key) { return m_bestMatches[key]; }
+    const T *Find_Best_Info(std::vector<T> const &vector, K const &key) const
+    {
+        auto iter = m_bestMatches.find(key);
+        const T *ret = nullptr;
+
+        if (iter != m_bestMatches.end()) {
+            ret = (*iter).second;
+        }
+
+        if (ret) {
```

Comment:
`!= nullptr`

Is it likely that the m_bestMatches might have a second that is nullptr? Just wondering if we can get rid of this additional nullptr check and return on L41.

Reply:
We should do it the way this was done originally to be safe.